### PR TITLE
Fix more init bugs.

### PR DESCRIPTION
**Pull request by TheBlueMatt**
This is a follow-on to #10758 to help move 10758 along. The first fixes a regression in master that was partially fixed in 10758, the second I'm not sure if its a regression or not, but its clearly a bug that should be fixed.

### CODE REVIEW

**Comment by morcos** [comment]
utACK 818bb29

Similar to #10758, I think its hard to reason about all the possible edge cases here, but this is a step in the right direction and I think on balance better code than not having it.

**Comment by TheBlueMatt** [comment]
Rebased and addressed some of the comments from @sdaftuar at #10758.

File: src/init.cpp [hunk 1]
```diff
@@ -1414,6 +1414,8 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
                 // LoadBlockIndex will load fTxIndex from the db, or set it if
                 // we're reindexing. It will also load fHavePruned if we've
                 // ever removed a block file from disk.
+                // Note that it also sets fReindex based on the disk flag!
+                // From here on out fReindex and fReset mean something different!
```
Comment (ryanofsky) [inline src/init.cpp]: In commit "Fix resume-of-reindex-after-restart"

Maybe make this more specific. You could say "From here on out fReindex may be true even if fReset is false (indicating the node was shut down mid-reindex)."

**Review by ryanofsky — COMMENTED** [review]
Low-confidence utACK e7539f864984740b80efc44e1a8970f4353ff066. Changes make sense, though I'm not extremely familiar with this code, so I don't have a lot of confidence that they couldn't potentially cause problems.

**Comment by sdaftuar** [comment]
I'm not sure of the first commit:
efac91e65 `Always wait for threadGroup to exit in bitcoind shutdown`

but utACK for the rest.

**Comment by laanwj** [comment]
utACK

**Pull request merged**